reframe/frontend/executors/__init__.py

```diff
@@ -132,61 +132,50 @@ def __init__(self, case, listeners=[]):
         self.zombie = False
 
         # Timestamps for the start and finish phases of the pipeline
-        self._timestamp = {}
+        self._timestamps = {}
+
+    def duration(self, phase):
+        # Treat pseudo-phases first
+        if phase == 'compile_complete':
+            t_start = 'compile_start'
+            t_finish = 'compile_wait_finish'
+        elif phase == 'run_complete':
+            t_start = 'run_start'
+            t_finish = 'wait_finish'
+        elif phase == 'total':
+            t_start = 'setup_start'
+            t_finish = 'pipeline_end'
+        else:
+            t_start  = f'{phase}_start'
+            t_finish = f'{phase}_finish'
 
-    @property
-    def duration(self, phase=None):
-        if phase:
-            try:
-                res = (self._timestamp[f'{phase}_finish'] -
-                       self._timestamp[f'{phase}_start'])
-            except KeyError:
-                res = None
-
-            return res
-
-        res = {}
-        for phase in ['setup', 'compile', 'compile_wait', 'run',
-                      'wait', 'sanity', 'performance', 'finalize',
-                      'cleanup']:
-            try:
-                d = (self._timestamp[f'{phase}_finish'] -
-                     self._timestamp[f'{phase}_start'])
-                res[phase] = d
-            except KeyError:
-                res[phase] = None
-
-        if 'compile_start' in self._timestamp:
-            if 'compile_wait_finish' in self._timestamp:
-                f_t = 'compile_wait_finish'
-            else:
-                f_t = 'compile_finish'
+        start = self._timestamps.get(t_start)
+        if not start:
+            return None
 
-            res['compile_complete'] = (self._timestamp[f_t] -
-                                       self._timestamp['compile_start'])
-        else:
-            res['compile_complete'] = None
+        finish = self._timestamps.get(t_finish)
+        if not finish:
+            finish = self._timestamps.get('pipeline_end')
 
-        if 'run_start' in self._timestamp:
-            if 'wait_finish' in self._timestamp:
-                f_t = 'wait_finish'
-            else:
-                f_t = 'run_finish'
+        return finish - start
 
-            res['run_complete'] = (self._timestamp[f_t] -
-                                   self._timestamp['run_start'])
-        else:
-            res['run_complete'] = None
+    def pipeline_timings(self, phases):
+        def _tf(t):
+            return f'{t:.3f}' if t else 'n/a'
 
-        try:
-            res['total'] = (
-                self._timestamp['pipeline_end'] -
-                self._timestamp['setup_start']
-            )
-        except KeyError:
-            res['total'] = None
+        mssg = ''
```

Comment:
Better use everywhere `msg` instead of `mssg`. This is the convention we use throughout the framework.